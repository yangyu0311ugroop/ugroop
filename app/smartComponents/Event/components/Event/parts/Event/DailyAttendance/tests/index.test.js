/**
 * Created by stephenkarpinskyj on 16/11/18.
 */

import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import { DailyAttendance } from '..';

describe('<DailyAttendance />', () => {
  let wrapper;
  let instance;

  const makeProps = () => ({});

  beforeEach(() => {
    wrapper = shallow(<DailyAttendance {...makeProps()} />);
    instance = wrapper.instance();
  });

  it('exists', () => {
    expect(DailyAttendance).toBeDefined();
  });

  describe('#renderPart()', () => {
    it('still matches snapshot', () => {
      const Component = () => <div />;
      expect(instance.renderPart(Component)()).toMatchSnapshot();
    });
  });

  describe('#render()', () => {
    it('not explodes', () => {
      expect(wrapper).toHaveLength(1);
    });

    it('still matches snapshot', () => {
      expect(toJSON(wrapper)).toMatchSnapshot();
    });
  });
});
