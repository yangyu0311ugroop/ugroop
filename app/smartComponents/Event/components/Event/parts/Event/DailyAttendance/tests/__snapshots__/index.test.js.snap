// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DailyAttendance /> #render() still matches snapshot 1`] = `
<ConnectFunction
  dataId={null}
  renderAccommodation={[Function]}
  subtype={null}
  type={null}
/>
`;

exports[`<DailyAttendance /> #renderPart() still matches snapshot 1`] = `
<Component
  dataId={null}
  formSubtype={null}
  formType={null}
  valuePath={
    Array [
      "attendance",
      "recordEachDay",
    ]
  }
/>
`;
